### ISSUE TITLE
1D Diffusion module

### ISSUE THREAD

**Issue by glwagner** [climate-machine/SingleColumnModels.jl#4]
The first module we build should probably be for solving the heat equation in one dimension:

```
∂t c = ∂z (K ∂z) c
```
where `K` can be a constant or variable.

We can use the development of this module to design shared abstractions and standard interfaces for user interaction with `SingleColumnModels`.

**Comment by charleskawczynski** [comment]
I agree. The heat equation is solved in the [EDMF PR](https://github.com/climate-machine/SingleColumnModels.jl/pull/3/files#diff-e34dbd75f283aa4e977e301e9fedffd4), but it would be nice to have more tests abstraction layers.

**Comment by glwagner** [comment]
Will you add examples that demonstrate a few simple set ups?

**Comment by charleskawczynski** [comment]
Yeah, that sounds like a good idea
